libp2p/pubsub/pubsub.py

```diff
@@ -294,6 +309,21 @@ def get_msg_validators(self, msg: rpc_pb2.Message) -> Tuple[TopicValidator, ...]
             # Force context switch
             await asyncio.sleep(0)
 
+    async def handle_dead_peer_queue(self) -> None:
+        """
+        Continuously read from dead peer queue and close the stream between that peer and
+        remove peer info from pubsub and pubsub router.
+        """
+        while True:
+
+            peer_id: ID = await self.dead_peer_queue.get()
+
+            # Remove Peer
+            asyncio.ensure_future(self._handle_dead_peer(peer_id))
+
+            # Force context switch
+            await asyncio.sleep(0)
```

Comment:
Why do you need this?  The context switch will happen as soon as the next iteration of the loop occurs and you start awaiting the next dead peer?